core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -52,18 +52,24 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
-  @GET
-  @Path("executors/{executorId}/threads")
-  def threadDump(@PathParam("executorId") execId: String): Array[ThreadStackTrace] = withUI { ui =>
+  private def checkExecutorId(execId: String): Unit = {
```

Comment:
If you don't mind, could you move these two new private methods, `checkExecutorId` and `checkAndGetSparkContext` at the end of this class in order to reduce the diff, please?

Reply:
thanks, @dongjoon-hyun, it looks much better